onnxruntime/core/graph/contrib_ops/contrib_defs.cc

```diff
@@ -431,6 +431,35 @@ and present state are optional. Present state could appear in output even when p
         return;
       });
 
+      static const char* Longformer_Attention_doc = R"DOC(
+Longformer Self Attention with a local context and a global context. Tokens attend locally: Each token
+attends to its W previous tokens and W succeding tokens with W being the window length. A selected few tokens
+attend globally to all other tokens.
+
+The attention mask is of shape (batch_size, sequence_length), where sequence_length is a multiple of 2W after padding.
+Mask value < 0 (like -10000.0) means the token is masked, 0 otherwise.
+
+Global attention flags have value 1 for the tokens attend globally and 0 otherwise.
+)DOC";
+
+  ONNX_CONTRIB_OPERATOR_SCHEMA(LongformerAttention)
+      .SetDomain(kMSDomain)
+      .SinceVersion(1)
+      .SetDoc(Longformer_Attention_doc)
+      .Attr("num_heads", "Number of attention heads", AttributeProto::INT)
+      .Attr("window", "One sided attention windows length W, or half of total window length", AttributeProto::INT)
+      .Input(0, "input", "3D input tensor with shape (batch_size, sequence_length, hidden_size), hidden_size = num_heads * head_size", "T")
+      .Input(1, "weight", "2D input tensor with shape (hidden_size, 3 * hidden_size)", "T")
+      .Input(2, "bias", "1D input tensor with shape (3 * hidden_size)", "T")
+      .Input(3, "mask", "Attention mask with shape (batch_size, sequence_length)", "T")
+      .Input(4, "global_weight", "2D input tensor with shape (hidden_size, 3 * hidden_size)", "T")
+      .Input(5, "global_bias", "1D input tensor with shape (3 * hidden_size)", "T")
+      .Input(6, "global", "Global attention flags with shape (batch_size, sequence_length)", "G")
+      .Output(0, "output", "3D output tensor with shape (batch_size, append_length, hidden_size)", "T")
+      .TypeConstraint("T", {"tensor(float)", "tensor(float16)"}, "Constrain input and output types to float tensors.")
+      .TypeConstraint("G", {"tensor(int32)"}, "Constrain to integer types")
+      .TypeAndShapeInferenceFunction(ONNX_NAMESPACE::propagateShapeAndTypeFromFirstInput);
+
```

Comment:
Do we need to add some check here?

Reply:
Checks can be found in https://github.com/microsoft/onnxruntime/blob/5ede0397f6deaff77d30faef5f5a8e9a0eed9207/onnxruntime/contrib_ops/cpu/bert/longformer_attention_base.cc#L19

Reply:
Got it